### DISCUSSION TITLE
Flagging Vods with muted segments

### DISCUSSION THREAD

**Discussion by Shockkota** [Zibbp/ganymede#404] [Ideas]
Would it be possible to start flagging vods with muted segments when flagged via the [api](https://dev.twitch.tv/docs/api/videos/). A later goal might be some automation defined on that, but just being able to easily identify them without having to search the vods manually would be nice.

Mainly looking at this from the perspective of purging a live archive if I have a full unmuted vod of the same stream.

**Comment by Zibbp** [comment]
I can create a table in the database, [similar to chapters](https://github.com/Zibbp/ganymede/blob/main/ent/schema/chapter.go), that stores all the muted segments. Unfortunately the official API doesn't return muted segments for some reason. Likely need more permissions for the channel. The unofficial GraphQL API does though, so that can be used to fetch the segments.
```json
   "data": {
      "video": {
        "id": "2115350511",
        "muteInfo": {
          "mutedSegmentConnection": {
            "nodes": [
              {
                "duration": 180,
                "offset": 8820,
                "__typename": "VideoMutedSegment"
              },
              {
                "duration": 180,
                "offset": 28080,
                "__typename": "VideoMutedSegment"
              },
              {
                "duration": 180,
                "offset": 29520,
                "__typename": "VideoMutedSegment"
              },
              {
                "duration": 180,
                "offset": 35460,
                "__typename": "VideoMutedSegment"
              },
              {
                "duration": 180,
                "offset": 35640,
                "__typename": "VideoMutedSegment"
              },
              {
                "duration": 180,
                "offset": 35820,
                "__typename": "VideoMutedSegment"
              },
              {
                "duration": 180,
                "offset": 42480,
                "__typename": "VideoMutedSegment"
              },
              {
                "duration": 180,
                "offset": 42840,
                "__typename": "VideoMutedSegment"
              },
              {
                "duration": 180,
                "offset": 46440,
                "__typename": "VideoMutedSegment"
              }
            ],
            "__typename": "VideoMutedSegmentConnection"
          },
          "__typename": "VideoMuteInfo"
        },
        "__typename": "Video"
      }
    },
```

Not sure how to best display the segments to the user. I can try to do some CSS with the player progress bar.

**Reply by Shockkota** [reply]
In my case its not even really knowing where the mute's are, though that could be useful. It's more of if a vod has no muted segments, then I would know it was "complete" archive and I could safely purge the live download so as to save on disk space from keeping both the live copy and the vod. I can likely script something to make the api calls myself to check and purge if needed. I just figured it was worth mentioning and considering making visible as I assume ganymede would see that flag when it made its calls for download.

**Reply by Zibbp** [reply]
I've merged https://github.com/Zibbp/ganymede/pull/406 into `main` which saves muted segments. If you want to give it a try switch the API image tag to `:main`. Muted segments are saved in a new database table and are available via the API `/api/v1/vod/VOD_ID?with_channel=true&with_chapters=true&with_muted_segments=true`. I haven't been able to find a great way to show it in the player yet so these are only available via the API and database.

**Reply by Zibbp** [reply]
Released in [v2.3.0](https://github.com/Zibbp/ganymede/releases/tag/v2.3.0)
